cpp/ql/src/semmle/code/cpp/ir/implementation/MemoryAccessKind.qll

```diff
@@ -43,6 +80,26 @@ class PhiMemoryAccess extends MemoryAccessKind, TPhiMemoryAccess {
   }
 }
 
+/**
+ * The operand is a ChiTotal operand, which accesses the same memory as its
+ * definition.
+ */
+class ChiTotalMemoryAccess extends MemoryAccessKind, TChiOldMemoryAccess {
+  override string toString() {
+    result = "chi(total)"
+  }
+}
+
+/**
+ * The operand is a ChiPartial operand, which accesses the same memory as its
+ * definition.
+ */
+class ChiPartialMemoryAccess extends MemoryAccessKind, TChiUpdateMemoryAccess {
```

Comment:
You renamed these classes from old/update to total/partial. I don't think that clarified anything for me, but the new comments on the member predicates of `ChiInstruction` did clarify it.

In any case, you now need `TChiOldMemoryAccess` and `TChiUpdateMemoryAccess` to be consistent with the types without a `T`.